code/modules/power/solar.dm

```diff
@@ -13,18 +13,33 @@
 	max_integrity = 150
 	integrity_failure = 0.33
 
-	var/id = 0
-	var/obscured = 0
-	var/sunfrac = 0
-	var/adir = SOUTH // actual dir
-	var/ndir = SOUTH // target dir
-	var/turn_angle = 0
-	var/obj/machinery/power/solar_control/control = null
+	var/id
+	var/obscured = FALSE
+	var/sunfrac = 0 //[0-1] measure of obscuration -- multipllier against power generation
+
+	var/azimuth_current = 0 //[0-360) degrees, which top-down direction are we facing?
+	var/elevation_current = 0 //[0-90] degrees, how high are we facing? negative numbers would be facing the floor
+	var/azimuth_target = 0 //same but what way we're going to face next time we turn
+	var/elevation_target = 45
+
+	var/obj/machinery/power/solar_control/control
+	var/needs_to_turn = TRUE //do we need to turn next tick?
+	var/needs_to_update_solar_exposure = TRUE //do we need to call update_solar_exposure() next tick?
+	//I should make these bitflags. Correction: You should make these bitflags.
+	var/list/turf/watched_turfs
+	var/obj/effect/overlay/panel
 
 /obj/machinery/power/solar/Initialize(mapload, obj/item/solar_assembly/S)
 	. = ..()
+	panel = new()
+	panel.icon = icon
+//	panel.vis_flags = VIS_INHERIT_ID|VIS_INHERIT_ICON|VIS_INHERIT_PLANE
```

Comment:
Just do 513 define ? Also any reason for this to be an object ? Doesn't seem to be interacting with anything.